config/config.go

```diff
@@ -41,6 +43,38 @@ func Config() *ConfigStruct {
 	return configData
 }
 
+//setMapValueEnvOverride recursively iterate over the keys and updates the map value accordingly
+func setMapValueEnvOverride(keys []string, mapValue map[string]interface{}, value interface{}) map[string]interface{} {
+	if len(keys) > 1  {
+		innerMapType := fmt.Sprintf("%T", mapValue[keys[0]])
+		if innerMapType  != "map[string]interface {}" {
+			log.Warn("cannot traverse full map path.  Unable to set ENV override. Returning ")
+			return mapValue
+		}
+		setMapValueEnvOverride(keys[1:], mapValue[keys[0]].(map[string]interface{}), value)
+	} else {
+		mapValue[keys[0]] = value
+	}
+
+	return mapValue
+}
+
+//applyEnvOverrides a bit of a hack to get around a viper limitation.
+// GetStringMap does not apply env overrides, so we have to traverse it again
+// and reset missing values
+func applyEnvOverrides(contexts map[string]interface{}, mapName string, config *viper.Viper) map[string]interface{} {
+	keys := config.AllKeys()
+	filteredKeys := funk.Filter(keys, func(s string) bool { return strings.Contains(s, "contexts.testing")})
```

Comment:
This is more of a personal preference, but I'm not a fan of these kinds of libraries that do away with type safety and have unclear function signatures. Especially in this case where you can achieve the same thing in the same number of lines:

for _, key := range config.AllKeys() {
  if !strings.Contains(s, "contexts.testing") {
    continue
  }
 value := config.Get(key)
  ...
}

Reply:
It creates cleaner code and already using funk in several places.